freqtrade/exchange/exchange.py

```diff
@@ -877,42 +877,67 @@ def get_dry_market_fill_price(self, pair: str, side: str, amount: float, rate: f
 
         return rate
 
-    def _is_dry_limit_order_filled(self, pair: str, side: str, limit: float) -> bool:
+    def _fill_dry_limit_order(self, pair: str, side: str,  # noqa: max-complexity: 13
+                              limit: float, given_amount: float) -> Tuple[Optional[float], float]:
+        """
+        Returns average price and filled amount
+        """
         if not self.exchange_has('fetchL2OrderBook'):
-            return True
-        ob = self.fetch_l2_order_book(pair, 1)
-        try:
+            return limit, given_amount
+        for order_book_top in (None, 1000):
+            ob = self.fetch_l2_order_book(pair, order_book_top)
             if side == 'buy':
-                price = ob['asks'][0][0]
-                logger.debug(f"{pair} checking dry buy-order: price={price}, limit={limit}")
-                if limit >= price:
-                    return True
+                obd = ob['asks']
             else:
-                price = ob['bids'][0][0]
-                logger.debug(f"{pair} checking dry sell-order: price={price}, limit={limit}")
-                if limit <= price:
-                    return True
-        except IndexError:
-            # Ignore empty orderbooks when filling - can be filled with the next iteration.
-            pass
-        return False
+                obd = ob['bids']
+            try:
+                logger.debug(
+                    f"{pair} checking dry {side}-order: "
+                    f"{limit=}, {given_amount=}, {order_book_top=}")
```

Comment:
I'll need to double-check if the minimum python version we're requiring did include this or not (3.8 at this point).

If it includes this - then this is fine. 
I remember that this used to "not work" on older python versions - but i forgot which version this was introduced :(

Reply:
It is brand new feature in 3.8
https://docs.python.org/3/whatsnew/3.8.html#f-strings-support-for-self-documenting-expressions-and-debugging

Reply:
Great, thanks for digging this out. I've been uncertain if this was added in 3.8 or 3.9.

however nothing in 3.8 is "brand new" anymore :laughing: